modules/jdbc/src/main/java/org/testcontainers/delegate/AbstractDatabaseDelegate.java

```diff
@@ -0,0 +1,61 @@
+package org.testcontainers.delegate;
+
+import java.util.Collection;
+
+/**
+ * @param <CONTAINER> testcontainers container
+ * @param <CONNECTION> connection to the database
+ * @author Eugeny Karpov
+ */
+public abstract class AbstractDatabaseDelegate<CONTAINER, CONNECTION> implements DatabaseDelegate {
+
+    /**
+     * Testcontainers container
+     */
+    protected CONTAINER container;
+
+    /**
+     * Database connection
+     */
+    private CONNECTION connection;
+
+    public AbstractDatabaseDelegate(CONTAINER container) {
+        this.container = container;
+    }
+
+    /**
+     * Get or create new connection to the database
+     */
+    protected CONNECTION getConnection() {
+        if (connection == null) {
+            this.connection = createNewConnection();
+        }
+        return connection;
+    }
+
+    @Override
+    public void execute(Collection<String> statements, String scriptPath, boolean continueOnError, boolean ignoreFailedDrops) {
+        try (DatabaseDelegate closeableDelegate = this) {
```

Comment:
why do we close our-self after "execute" call? O_O

Reply:
To close the connection after the statements execution. But it's a bad practice. Removed.